### PR TITLE
Simplify COLR API for transforms

### PR DESCRIPTION
**Pull request by LaurenzV**
Will make it easier to port code from harfbuzz to rustybuzz. I hope I did the `libm` thing correctly.

### CODE REVIEW

**Comment by LaurenzV** [comment]
Hmm... yeah some advice on how to deal with the feature thingy would be appreciated. 😄

File: Cargo.toml [hunk 1]
```diff
@@ -12,9 +12,15 @@ readme = "README.md"
...
 
+[dependencies]
+libm = { version = "0.2.8", optional = true }
+
 [features]
 default = ["std", "opentype-layout", "apple-layout", "variable-fonts", "glyph-names"]
+# Enables the use of the standard library. Deactivate this and activate the no-std
+# feature to compile for targets that don't have std.
 std = []
+no-std = ["libm"]
```
Comment (RazrFalcon) [inline Cargo.toml:23]: I would suggest using `no-std-float`, just like in `tiny-skia`.

File: src/lib.rs [hunk 2]
```diff
@@ -432,6 +432,44 @@ impl Transform {
         Transform::new(1.0, 0.0, 0.0, 1.0, tx, ty)
     }
 
+    /// Creates a new rotation transform.
+    #[inline]
+    pub fn new_rotate(angle: f32) -> Self {
+        #[cfg(feature = "no-std")]
```
Comment (RazrFalcon) [inline src/lib.rs:438]: I would suggest going the `tine-skia` route by using [a trait](https://github.com/RazrFalcon/tiny-skia/blob/784569522f728cc70232cbb88abd7330571b51dd/path/src/scalar.rs#L78-L125).

And then 
```rust
#[cfg(all(not(feature = "std"), feature = "no-std-float"))]
use crate::NoStdFloat;
```

where needed.

Comment (RazrFalcon) [reply src/lib.rs:438]: You can put the new trait somewhere in `lib.rs`

**Comment by LaurenzV** [comment]
I'll mark this as draft since I also want to test it in `rustybuzz` first, which might take a while.

**Comment by LaurenzV** [comment]
Is it okay that when building with no default features and `variable-fonts`, you must also add `no-std-float` (or std)? Kinda annoying, but not sure if there is a better way...

**Comment by RazrFalcon** [comment]
Yes, same with `tiny-skia`. `--no-default-features` implies adding `--feature libm`. No better way to handle this to my knowledge.

**Comment by LaurenzV** [comment]
Okay, should be good now.

File: Cargo.toml [hunk 3]
```diff
@@ -12,9 +12,15 @@ readme = "README.md"
 edition = "2018"
 exclude = ["benches/**"]
 
+[dependencies]
+libm = { version = "0.2.8", optional = true }
+
 [features]
 default = ["std", "opentype-layout", "apple-layout", "variable-fonts", "glyph-names"]
+# Enables the use of the standard library. Deactivate this and activate the no-std
```
Comment (RazrFalcon) [inline Cargo.toml:20]: `no-std-float`, not `no-std`

**Pull request merged**